src/Microsoft.AspNetCore.SignalR.Redis/RedisHubLifetimeManager.cs

```diff
@@ -399,7 +402,9 @@ public override Task OnDisconnectedAsync(HubConnectionContext connection)
                 throw new ArgumentNullException(nameof(groupName));
             }
 
-            if (await RemoveGroupAsyncCore(connectionId, groupName))
+
+            var connection = _connections[connectionId];
+            if (connection != null && await RemoveGroupAsyncCore(connection, groupName))
```

Comment:
Is this correct - if the connection is local (i.e. connection is not null) but the group does not exist we will send to other servers. Shouldn't this just be:

```
if (connection != null)
{
    await RemoveGroupAsyncCore(connection, groupName);
    return;
}
```
?

Reply:
Good catch